cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/validation/StartupValidator.java

```diff
@@ -33,18 +34,18 @@
 public final class StartupValidator
 {
     private static final Logger LOGGER = LoggerFactory.getLogger(StartupValidator.class);
-    private static final StartupValidator INSTANCE = new StartupValidator();
+    private static final ThreadLocal<StartupValidator> INSTANCE = ThreadLocal.withInitial(StartupValidator::new);
     private static final String DISABLE = "SKIP_STARTUP_VALIDATIONS";
 
-    private final List<StartupValidation> validations = new ArrayList<>();
+    private final List<StartupValidation> validations = Collections.synchronizedList(new ArrayList<>());
```

Comment:
`CopyOnWriteArrayList` might be more suitable?

Reply:
The fact of the matter is that it's now wrapped in a ThreadLocal so it shouldn't really _need_ to be synchronized anyway, but we decided to do this as an additional guarantee to make sure that it couldn't really be updated from multiple threads... the number of accesses to this validation collection is like 3 times per thread so I think this is reasonable.

Reply:
Meh - I'll change it... we do end up iterating over it so who knows.